common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivityFactory.java

```diff
@@ -0,0 +1,128 @@
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.AuthorizationIntentKey.WEB_VIEW_ZOOM_CONTROLS_ENABLED;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.AuthorizationIntentKey.WEB_VIEW_ZOOM_ENABLED;
+
+/**
+ * Constructs intents and/or fragments for interactive requests based on library configuration and current request.
+ */
+public class AuthorizationActivityFactory {
+
+    /**
+     * Return the correct authorization activity based on library configuration.
+     *
+     * @param context                    Android application context
+     * @param authIntent                 Android intent used by the authorization activity to launch the specific implementation of authorization (BROWSER, EMBEDDED)
+     * @param requestUrl                 The authorization request in URL format
+     * @param redirectUri                The expected redirect URI associated with the authorization request
+     * @param requestHeaders             Additional HTTP headers included with the authorization request
+     * @param authorizationAgent         The means by which authorization should be performed (EMBEDDED, WEBVIEW) NOTE: This should move to library configuration
+     * @param webViewZoomEnabled         This parameter is specific to embedded and controls whether webview zoom is enabled... NOTE: Needs refactoring
+     * @param webViewZoomControlsEnabled This parameter is specific to embedded and controls whether webview zoom controls are enabled... NOTE: Needs refactoring
+     * @return An android Intent which will be used by Android to create an AuthorizationActivity
+     */
+    public static Intent getAuthorizationActivityIntent(final Context context,
+                                                        final Intent authIntent,
+                                                        final String requestUrl,
+                                                        final String redirectUri,
+                                                        final HashMap<String, String> requestHeaders,
+                                                        final AuthorizationAgent authorizationAgent,
+                                                        final boolean webViewZoomEnabled,
+                                                        final boolean webViewZoomControlsEnabled) {
+        Intent intent;
+        final LibraryConfiguration libraryConfig = LibraryConfiguration.getInstance();
+        if (ProcessUtil.isBrokerProcess(context)) {
+            intent = new Intent(context, BrokerAuthorizationActivity.class);
+        } else {
+            if (libraryConfig.isAuthorizationInCurrentTask()) {
+                intent = new Intent(context, CurrentTaskAuthorizationActivity.class);
+            } else {
+                intent = new Intent(context, AuthorizationActivity.class);
+            }
+        }
```

Comment:
I believe this should be equivalent 

```suggestion
        } else if (libraryConfig.isAuthorizationInCurrentTask()) {
                intent = new Intent(context, CurrentTaskAuthorizationActivity.class);
        } else {
                intent = new Intent(context, AuthorizationActivity.class);
        }
```